lib/kdump_utils.pm

```diff
@@ -171,6 +179,7 @@ sub activate_kdump {
     }
     send_key('alt-o');
     if ($expect_restart_info == 1) {
+        handle_warning_install_os_prober();
```

Comment:
That adds a fixed 200s delay even in cases where that screen would never show up (because e.g. os-prober had been installed previously)

Reply:
you are right. i forgot to remove the timeout.

Reply:
removed

Reply:
without specifying a timeout, the default of 30s is used.

It's probably better to combine the possible matches in one assert screen, and then check based on which one we had to decide if the install is needed or not.

something along this (pseudo code)

```
loop
  assert_screen os-prober-warning yast2-kdump-restart-info
  if tag_has_match os-prober-warning: perform installation
until tag_has_match yast2-kdump-restart-info
```

Reply:
Correct me if i am wrong but i think it is not necessary in this case. the needle is at an expected moment. if the condition   does not match it will just continue(with the default timeout delay). isnt it more expensive to introduce a loop here?

Reply:
ignore. i see based on the pseudo code again that the `yast2-kdump-restart-info` will have to be checked first. will do that. thanks

Reply:
http://aquarius.suse.cz/tests/6480# uses the loop
http://aquarius.suse.cz/tests/6479 is with the submitted code.
the first is ~30s slower.

Reply:
and the time difference is in forced_scheduled_tasks - definitively not related to this loop.

the module we actually look at, kdump_and_crash, was ~ 10s faster in t6480

Reply:
submitted. thanks and let me know. @DimStar77